test/function/samples/handle-statically-resolvable-if-statements/main.js

```diff
@@ -0,0 +1,7 @@
+export let x = 0;
+
+if (value === 1) {
+	x = 1;
+} else if (12 !== 12) {
+	x = 2;
+}
```

Comment:
I found myself really wanting to be able to review the output on this one - form tests would be preferable once we've got these set up on a single module format for these analysis scenarios.

Reply:
Note that for `function` tests, you can always add `show: true` to the config to inspect the output.

Reply:
Yes I understand, function is super important to catch things too. I wanted to check the `JSON.stringify(getValue()` case tests which this seemed to be and it helps to review to see the cases without having to clone. So that's why I was wondering if getting those analysis form test structures going would help here.

Reply:
BTW the stringify/overwrite logic does in fact not cover your case. The only situation where this logic is applied at the moment is if both branches are included but the test is not included. This is *only*  possible if
* the statement is statically resolvable, but
* the dead branch is included because it contains a used variable declaration

This would be an example:
https://rollupjs.org/repl?version=0.58.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGxldCUyMHglMjAlM0QlMjAwJTNCJTVDbiU1Q25pZiUyMCgxMiUyMCUzRCUzRCUzRCUyMDEyKSUyMCU3QiU1Q24lNUN0eCUyMCUzRCUyMDElM0IlNUNuJTdEJTIwZWxzZSUyMCU3QiU1Q24lNUN0dmFyJTIweSUzQiU1Q24lN0QlNUNuJTVDbmV4cG9ydCUyMCU3QnklN0QlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

In all other situations, either the test is not rendered at all or it needs to be rendered as it is because the value is unknown.

Reply:
Interesting, yes that fills in the gap I was missing thanks.